modules/@apostrophecms/schema/lib/addFieldTypes.js

```diff
@@ -752,6 +752,19 @@ module.exports = (self) => {
       if ((field.max !== undefined) && (results.length > field.max)) {
         throw self.apos.error('max');
       }
+      if (data.length && field.schema && field.schema.length) {
+        const { name: uniqueFieldName, label: uniqueFieldLabel } = field.schema.find(subfield => subfield.unique);
+        if (uniqueFieldName) {
+          const duplicates = data
+            .map(item => Array.isArray(item[uniqueFieldName])
+              ? item[uniqueFieldName][0]._id
+              : item[uniqueFieldName])
+            .filter((item, index, array) => array.indexOf(item) !== index);
+          if (duplicates.length) {
```

Comment:
we might need to find a way to flatten this levels of nested if statements

Reply:
Open to ideas